src/components/learner-credit-management/BudgetDetailPageOverviewAvailability.jsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button, Col, Hyperlink, ProgressBar, Row, Stack,
+} from '@edx/paragon';
+import { Add } from '@edx/paragon/icons';
+import { generatePath, useRouteMatch } from 'react-router-dom';
+import { useHistory } from 'react-router';
+import { formatPrice } from './data';
+import { configuration } from '../../config';
+
+const BudgetDetail = ({ available, utilized, limit }) => {
+  const currentProgressBarLimit = (available / limit) * 100;
+
+  return (
+    <Stack className="border border-light-400 p-4">
+      <h4>Available</h4>
+      <Stack direction="horizontal" gap={4} className="mt-1">
+        <span className="display-1 text-dark" data-testid="budget-detail-available">{formatPrice(available)}</span>
+        <span className="mt-auto small text-monospace" data-testid="budget-detail-utilized">
+          Utilized {formatPrice(utilized)}
+        </span>
+      </Stack>
+      <Stack gap={2} className="mt-3">
+        <ProgressBar now={currentProgressBarLimit} variant="info" />
+        <span className="ml-auto small text-monospace" data-testid="budget-detail-limit">
+          {formatPrice(limit)} limit
+        </span>
+      </Stack>
+    </Stack>
+  );
+};
+
+BudgetDetail.propTypes = {
+  available: PropTypes.number.isRequired,
+  utilized: PropTypes.number.isRequired,
+  limit: PropTypes.number.isRequired,
+};
+
+const BudgetActions = ({ budgetId, isAssignable }) => {
+  const routeMatch = useRouteMatch();
+  const history = useHistory();
+  const supportUrl = configuration.ENTERPRISE_SUPPORT_URL;
+
+  if (!isAssignable) {
+    return (
+      <div className="p-4">
+        <h4>Get people learning using this budget</h4>
+        <p>
+          Funds from this budget are set auto allocate to registered
+          learners based on settings configured with your support team.
+        </p>
+        <Button variant="outline-primary" as={Hyperlink} destination={supportUrl} target="_blank">
+          Contact support
+        </Button>
+      </div>
+    );
+  }
+
+  const handleSkipToCatalog = (key) => {
+    const location = {
+      pathname: generatePath(routeMatch.path, { budgetId, activeTabKey: 'catalog' }),
+      state: { budgetActivityScrollToKey: key },
+    };
+
+    history.push(location);
+  };
+
+  return (
+    <div className="p-4">
+      <h4>Get people learning using this budget</h4>
+      <Button
+        variant="brand"
+        className="mt-3"
+        iconBefore={Add}
+        onClick={() => handleSkipToCatalog('catalog')}
```

Comment:
Related, it looks like when I'm already on the "Catalog" tab route and click this CTA button (in this current implement or the suggested implementation with `Link`), it doesn't seem to scroll down to the "Catalog" tab contents. Do you also see this behavior?

Reply:
It appears that the `BudgetDetailCatalogTabContents` doesn't re-render when already on the catalog route, which in turn prevents the triggering of the scrollToView functionality within the component.